integration/src/integration/java/pl/allegro/tech/hermes/integration/management/ListUnhealthySubscriptionsForOwnerBasedOnGraphiteTest.java

```diff
@@ -0,0 +1,387 @@
+import pl.allegro.tech.hermes.integration.helper.GraphiteEndpoint;
+import pl.allegro.tech.hermes.integration.helper.Waiter;
+import pl.allegro.tech.hermes.integration.setup.HermesManagementInstance;
+import pl.allegro.tech.hermes.test.helper.endpoint.HermesAPIOperations;
+import pl.allegro.tech.hermes.test.helper.endpoint.HermesEndpoints;
+import pl.allegro.tech.hermes.test.helper.util.Ports;
+
+import java.util.List;
+
+import static jakarta.ws.rs.core.MediaType.TEXT_PLAIN;
+import static org.assertj.core.api.AssertionsForClassTypes.catchThrowable;
+import static pl.allegro.tech.hermes.api.SubscriptionHealthProblem.malfunctioning;
+import static pl.allegro.tech.hermes.integration.env.SharedServices.services;
+import static pl.allegro.tech.hermes.integration.helper.GraphiteEndpoint.subscriptionMetricsStub;
+import static pl.allegro.tech.hermes.integration.test.HermesAssertions.assertThat;
+import static pl.allegro.tech.hermes.test.helper.builder.SubscriptionBuilder.subscription;
+import static pl.allegro.tech.hermes.test.helper.builder.TopicBuilder.randomTopic;
+import static pl.allegro.tech.hermes.test.helper.builder.TopicBuilder.topic;
+
+public class ListUnhealthySubscriptionsForOwnerBasedOnGraphiteTest extends IntegrationTest {
+
+    private final Client httpClient = ClientBuilder.newClient();
+    private GraphiteEndpoint graphiteEndpoint;
+    private HermesAPIOperations apiOperations;
+    private String managementUrl;
+    private HermesEndpoints customManagement;
+    private Waiter hermesWaiter;
+
+    @BeforeClass
+    public void initializeOnce() {
+        int managementPort = Ports.nextAvailable();
+        managementUrl = "http://localhost:" + managementPort + "/";
+        HermesManagementInstance hermesManagement = HermesManagementInstance.starter()
+                .port(managementPort)
+                .withGraphiteExternalStorageEnabled()
+                .addKafkaCluster(DC1, kafkaClusterOne.getBootstrapServersForExternalClients())
+                .addZookeeperCluster(DC1, hermesZookeeperOne.getConnectionString())
+                .schemaRegistry(schemaRegistry.getUrl())
+                .replicationFactor(kafkaClusterOne.getAllBrokers().size())
+                .uncleanLeaderElectionEnabled(false)
+                .start();
+        apiOperations = hermesManagement.operations();
+        customManagement = new HermesEndpoints(managementUrl, CONSUMER_ENDPOINT_URL);
+        hermesWaiter = new Waiter(customManagement, services().zookeeper(), brokerOperations, PRIMARY_KAFKA_CLUSTER_NAME, KAFKA_NAMESPACE);
+    }
+
+    @BeforeMethod
+    public void initializeAlways() {
+        graphiteEndpoint = new GraphiteEndpoint(SharedServices.services().graphiteHttpMock());
+    }
+
+    @AfterMethod
+    public void cleanup() {
+        removeSubscriptions(customManagement, hermesWaiter);
+        removeTopics(customManagement, hermesWaiter);
+    }
+
+    @Test
+    public void shouldNotListHealthySubscriptions() {
```

Comment:
The test means: "if the subscription is healthy, don't list it in the unhealthy subscription endpoint". If it would be unhealthy it should be definitely listed :)